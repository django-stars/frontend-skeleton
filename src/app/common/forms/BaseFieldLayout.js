import PropTypes from 'prop-types'
// import TooltipIcon from 'shared/widgets/TooltipIcon'

BaseFieldLayout.propTypes = {
  icon: PropTypes.node,
  label: PropTypes.string,
  prefix: PropTypes.string,
  // tooltipContent: PropTypes.string,
  required: PropTypes.bool,
  InputComponent: PropTypes.element,
}

export default function BaseFieldLayout(props) {
  const {
    icon,
    label,
    prefix,
    // tooltipContent,
    required,
    inputComponent: InputComponent,
  } = props

  return (
    <div className='form-group'>
      {icon}
      {label && (
        <label className='control-label'>
          {label}
          {required && <span className='control-asterisk'>*</span>}
        </label>
      )}
      <div className='control-field'>
        <div className='control-element'>
          {prefix && <div className='control-prefix'>{prefix}</div>}
          <InputComponent
            {...props}
            {...props.input}
          />
          {props.meta.error}
          {/* tooltipContent && <TooltipIcon {...props} /> */}
        </div>
      </div>
    </div>
  )
}
