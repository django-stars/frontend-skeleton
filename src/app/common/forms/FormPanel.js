import PropTypes from 'prop-types'
import { Panel } from 'react-bootstrap'

const propTypes = {
  panelTitle: PropTypes.string,
  isCollapsible: PropTypes.bool,
  isExpanded: PropTypes.bool,
}

const defaultProps = {
  isCollapsible: false,
  isExpanded: true,
}

export default function FormPanel(props) {
  const { panelTitle, isCollapsible, isExpanded, ...restProps } = props
  return (
    <Panel defaultExpanded={isExpanded}>
      <Panel.Heading>
        <Panel.Title componentClass='h3' toggle={isCollapsible}>{panelTitle}</Panel.Title>
        { isCollapsible && (<Panel.Toggle componentClass='i' className='collapse-icon'></Panel.Toggle>) }
      </Panel.Heading>
      <Panel.Body collapsible={isCollapsible} {...restProps} />
    </Panel>
  )
}

FormPanel.propTypes = propTypes
FormPanel.defaultProps = defaultProps
