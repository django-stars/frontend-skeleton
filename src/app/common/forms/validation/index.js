import { compose, composeValidators, mainValidation } from './utils'
import email from './email'
import required from './required'

export function validateEmail(fields) {
  return mainValidation(fields, email)
}

export function validateRequired(fields) {
  return mainValidation(fields, required)
}

export {
  compose,
  composeValidators,
  email,
  required,
}
