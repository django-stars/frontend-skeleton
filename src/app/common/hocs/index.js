import prefetchResources from './prefetchResources'
import withReduxForm from './withReduxForm'

export {
  prefetchResources,
  withReduxForm,
}
