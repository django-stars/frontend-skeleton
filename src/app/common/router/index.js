import RouteRecursive from './RouteRecursive'
import navigate from './navigate'
import { Link, NavLink } from './Link'
import match from './match'
// import Prompt from './Prompt' TODO: remove commented code if it is not necessary, sebastiyan, 20.03.2018

export {
  RouteRecursive,
  navigate,
  Link,
  NavLink,
  match,
  // Prompt,
}
