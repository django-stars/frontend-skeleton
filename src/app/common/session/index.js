import CheckAccess from './CheckAccess'
export * as access from './access'

const logout = function() {
  return {
    type: '@ds-resource/set-data',
    meta: {
      resource: { namespace: 'session' },
    },
    payload: {},
  }
}

export {
  CheckAccess,
  logout,
}
