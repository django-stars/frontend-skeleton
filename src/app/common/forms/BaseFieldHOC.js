import { Field } from 'redux-form'
import BaseFieldLayout from './BaseFieldLayout'

export default function BaseFieldHOC(Component) {
  return function(props) {
    return (
      <Field
        component={BaseFieldLayout}
        inputComponent={Component}
        {...props}
      />
    )
  }
}
