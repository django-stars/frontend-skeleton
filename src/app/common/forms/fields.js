import BaseFieldHOC from './BaseFieldHOC'

import CheckboxInput from './inputs/CheckboxInput'
// import CurrencyInput from './inputs/CurrencyInput'
// import DateInput from './inputs/DateInput'
import NumberInput from './inputs/NumberInput'
import RadiosInput from './inputs/RadiosInput'
// import SelectInput from './inputs/SelectInput'
// import SelectRangeInput from './inputs/SelectRangeInput'
import TextInput from './inputs/TextInput'
import TextAreaInput from './inputs/TextAreaInput'


const CheckboxField = BaseFieldHOC(CheckboxInput)
// const CurrencyField = BaseFieldHOC(CurrencyInput)
// const DateField = BaseFieldHOC(DateInput)
const NumberField = BaseFieldHOC(NumberInput)
const RadiosField = BaseFieldHOC(RadiosInput)
// const SelectField = BaseFieldHOC(SelectInput)
// const SelectRangeField = BaseFieldHOC(SelectRangeInput)
const TextField = BaseFieldHOC(TextInput)
const TextAreaField = BaseFieldHOC(TextAreaInput)

export {
  CheckboxField,
  // CurrencyField,
  // DateField,
  NumberField,
  RadiosField,
  // SelectField,
  // SelectRangeField,
  TextField,
  TextAreaField,
}
