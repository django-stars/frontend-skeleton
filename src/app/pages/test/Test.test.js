import { shallow } from 'enzyme'
import { shallowToJson } from 'enzyme-to-json'

import { TextField } from 'common/forms'
import Test from './Test'

describe('component Test', function() {
  it('should match snapshot', function() {
    const component = shallow(<Test />)

    expect(shallowToJson(component)).toMatchSnapshot()
  })

  it('should have only one TextField with name "firstName"', function() {
    const textField = shallow(<Test />).find(TextField).filter({ name: 'firstName' })

    expect(textField.length).toEqual(1)
  })

  it('should not contain h2 headline with text', function() {
    const wrapper = shallow(<Test />)
    const text = <h2>Text to test</h2>

    expect(wrapper.contains(text)).toEqual(false)
  })
})
