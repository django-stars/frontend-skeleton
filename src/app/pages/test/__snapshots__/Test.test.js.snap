// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component Test should match snapshot 1`] = `
<form>
  <Component
    name="firstName"
  />
  <Component
    name="lastName"
  />
  <Component
    name="salutation"
    options={
      Array [
        Object {
          "label": "Mrs.",
          "value": "mrs",
        },
        Object {
          "label": "Mr.",
          "value": "mr",
        },
      ]
    }
  />
  <button
    className="btn btn-defaut"
    type="submit"
  >
    submit to console
  </button>
</form>
`;
